carbonmark/components/pages/Project/ProjectView/ListView.tsx

```diff
@@ -0,0 +1,172 @@
+  {
+    id: "name",
+    label: "Name",
+    sortable: false,
+    disablePadding: true,
+  },
+  {
+    id: "price",
+    label: "Price",
+    sortable: true,
+    disablePadding: false,
+  },
+  {
+    id: "description",
+    label: "Description",
+    sortable: false,
+    disablePadding: false,
+  },
+  {
+    id: "category",
+    label: "Category",
+    sortable: false,
+    disablePadding: false,
+  },
+  {
+    id: "vintage",
+    label: "Vintage",
+    sortable: true,
+    disablePadding: false,
+  },
+];
+
+type Props = {
+  projects: Array<Project>;
+  form: UseFormReturn<FilterValues>;
+};
+
+export const ListView: FC<Props> = ({ form, projects }) => {
+  const router = useRouter();
+  const sort = form.watch("sort");
+  const [sortOrder, setSortOrder] = useState<"asc" | "desc">("asc");
+  const [sortColumn, setSortColumn] = useState<keyof Data>();
+  const sortDirection = sortOrder ?? "desc";
+
+  const handleSort = (column: keyof Data) => {
+    setSortColumn(column);
+    setSortOrder(sortOrder === "asc" ? "desc" : "asc");
+  };
+
+  useEffect(() => {
+    // when the user loads the projects view with some previously set sort value the sort
+    // order should be reflected in the table header with the correct arrow direction
+    const [column, order] = split(sort, "-");
+    setSortColumn(column as keyof Data);
+    setSortOrder(["newest", "highest"].includes(order) ? "asc" : "desc");
+  }, [sort]);
+
+  useEffect(() => {
+    // updates the sort dropdown when there is an update to the table sorting
+    if (sortColumn === "vintage") {
+      form.setValue(
+        "sort",
+        isEqual(sortOrder, "asc") ? "vintage-newest" : "vintage-oldest"
+      );
+    }
+    if (sortColumn === "price") {
+      const sort = sortOrder === "asc" ? "price-highest" : "price-lowest";
+      form.setValue("sort", sort);
+    }
+  }, [sortOrder]);
+
+  return (
+    <TableContainer className={styles.listView} component={Paper}>
+      <Table>
+        <TableHead>
+          <TableRow>
+            {columns.map((column) => (
+              <TableCell
+                align="left"
+                key={column.id}
+                sortDirection={sortDirection}
+                padding={column.disablePadding ? "none" : "normal"}
+              >
+                <TableSortLabel
+                  direction={sortDirection}
+                  hideSortIcon={!column.sortable}
+                  onClick={() => handleSort(column.id)}
+                  active={column.sortable && column.id === sortColumn}
+                >
+                  {column.label}
+                </TableSortLabel>
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {projects.map((project: Project, index: number) => (
+            <TableRow
+              hover
+              tabIndex={-1}
+              key={`${project.key}-${index}`}
+              onClick={() => router.push(createProjectLink(project))}
```

Comment:
I did think about this but opted for using the onClick handler as the `Link` element wraps an anchor tag around the table row making it semantically incorrect. I also thought about individually adding a link for each `td` item but I thought it would just add more cluttered code. Happy to hear your thoughts, maybe your aware of another approach when using tables with `Link`?.